src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneIntegrationTest.java

```diff
@@ -258,6 +260,102 @@ public void byIdNotFound() {
           .body("errors", is(nullValue()));
     }
 
+    @Test
+    public void inCondition() {
+      String json =
+          """
+        {
+          "findOne": {
+            "filter" : {"_id" : {"$in": ["doc5", "doc4"]}}
+          }
+        }
+        """;
+      // findOne resolves any one of the resolved documents. So the order of the documents in the
+      // $in clause is not guaranteed.
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.count", is(1))
+          .body("data.docs", hasSize(1))
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.docs[0]", anyOf(jsonEquals(DOC5_JSON), jsonEquals(DOC4_JSON)));
+    }
+
+    @Test
+    public void inConditionEmptyArray() {
+      String json =
+          """
+        {
+          "findOne": {
+            "filter" : {"_id" : {"$in": []}}
+          }
+        }
+            """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("errors", is(notNullValue()))
+          .body("errors[1].message", is("$in operator must have at least one value"))
```

Comment:
Is it necessary to throw an error in this case, rather than simply returning no results? MongoDB just returns empty array:

```
MongoDB Enterprise > db.tests.insertOne({ x: 1 })
{
	"acknowledged" : true,
	"insertedId" : ObjectId("643b4b87e043b29280bd8ff5")
}
MongoDB Enterprise > db.tests.find({ x: { $in: [] } })
MongoDB Enterprise > 
```